runner/copy.go

```diff
@@ -75,28 +78,64 @@ func (m *MaterializedChunkRunner) Run() (types.CompletionWatcher, error) {
 			jobWatcher.EndWatch(fmt.Errorf("failed to create iterator: %w", err))
 			return
 		}
-		i := 0
-		for it.Next() {
-			i += 1
-			value := it.Value().Value
-			entity := it.Value().Entity
-			err := m.Table.Set(entity, value)
-			if err != nil {
-				jobWatcher.EndWatch(fmt.Errorf("could not set table: %w", err))
-				return
+		// Buffer the records channel to ensure readers won't block the writer loop below
```

Comment:
I'd add a comment above all of this to explain the general algorithm:

One reader thread reads as fast as it can and writers to a buffered channel. workerSize go routines run and each reads from the buffered channel and writes to Redis, waiting for a response.